packages/evm/src/eof/container.ts

```diff
@@ -269,6 +269,27 @@ class EOFHeader {
     }
     return offset
   }
+
+  // Returns the code section for a given program counter position
+  getSectionFromProgramCounter(programCounter: number) {
+    if (
+      programCounter < 0 ||
+      programCounter >
+        this.codeStartPos[this.codeStartPos.lastIndex] + this.codeSizes[this.codeSizes.lastIndex]
```

Comment:
This throws if `codeStartPos` is not initialized for the last position, right? It will only be initialized once we start to CALLF / JUMPF to this final code section

Reply:
I think in general this method can only be called if the `codeStartPos` / `codeSizes` is properly initialized (cached)

Reply:
I tested this locally and it works perfectly if I make this fix.  Will push a commit along with a simple test to validate.

Reply:
Yup, that makes sense!

Reply:
Sorry, misread, have posted a lengthy point about this at the newly added line. The problem is that the on-the-fly caching mechanism is not optimized anymore if we initialize the entire cache at the container instantiation each time.

Reply:
See: https://github.com/ethereumjs/ethereumjs-monorepo/pull/3953/commits/62ae8fd90035755c75a07494b68712db547eaf19#r2056935051